mpactpy/pinmesh.py

```diff
@@ -14,6 +14,195 @@
+            mat_specs : any
+                Material specifications or mapping information needed for material assignment.
+
+    Returns
+    -------
+    List[Material]
+        A list of Material objects corresponding to each centroid in the batch.
+    """
+    points_batch, model_data, mat_specs = args
+
+    model = _reconstruct_openmc_model(model_data)
+
+    results = []
+    for point in points_batch:
+        mat = Material.from_openmc_model_point(point, model, mat_specs)
+        results.append(mat)
+    return results
+
+
+def _process_homogenized_batch(args) -> List[Material]:
+    """ Processes a batch of centroid points to determine material assignments in parallel.
+
+    Parameters
+    ----------
+    args : tuple
+        A tuple containing:
+            points_batch : list of tuple
+                A list of points (e.g., coordinates) to be processed.
+            model_data : any
+                Serialized data required to reconstruct the OpenMC model.
+            mat_specs : any
+                Material specifications or mapping information needed for material assignment.
+
+    Returns
+    -------
+    List[Material]
+        A list of Material objects corresponding to each element in the batch.
+    """
+    elements_batch, model_data, mat_specs, mix_policy = args
+
+    model = _reconstruct_openmc_model(model_data)
+
+    results = []
+    for element in elements_batch:
+        mat = Material.from_openmc_model_element(element, model, mat_specs, mix_policy)
+        results.append(mat)
+    return results
+
+
+def _materials_at_centroids(centroids: np.ndarray,
+                            model: openmc.Model,
+                            overlay_policy: PinMesh.OverlayPolicy) -> List[Material]:
+    """ Determines material assignments at specified centroids
+
+    Parameters
+    ----------
+    centroids : numpy.ndarray
+        Array of centroid coordinates to process.
+    model : openmc.Model
+        The OpenMC model used to determine material assignments.
+    overlay_policy : PinMesh.OverlayPolicy
+        Policy object specifying overlay options.
+
+    Returns
+    -------
+    List[Material]
+        A list of Material objects corresponding to each centroid.
+    """
+    chunk_size = max(1, len(centroids) // overlay_policy.num_procs)
+    chunks     = [centroids[i:i+chunk_size] for i in range(0, len(centroids), chunk_size)]
+
+    # Model serialization required for multiprocessing
+    model_data = _serialize_openmc_model(model)
+
+    # Prepare arguments for multiprocessing
+    args_list = [(chunk, model_data, overlay_policy.mat_specs) for chunk in chunks]
+
+    with ProcessPoolExecutor(max_workers=overlay_policy.num_procs) as executor:
+        batch_results = list(executor.map(_process_centroid_batch, args_list))
+
+    # Flatten results back to original order
+    materials = []
+    for batch_result in batch_results:
+        materials.extend(batch_result)
```

Comment:
```suggestion
materials = [item for sublist in batch_results for item in sublist]
```
If you want, you can simplify the flattening.  These lines appears a couple times in the code.